lib/game.rb

```diff
@@ -0,0 +1,30 @@
+require_relative 'scorecard'
+require_relative 'frame'
+require_relative 'roll'
+
+class Game
+  attr_reader :scorecard, :current_frame, :frames, :roll_count
+
+  def initialize
+    @scorecard = Scorecard.new
+    @current_frame = Frame.new
+    @frames = [@current_frame]
+    @roll_count = 0
```

Comment:
I like the logic of splitting the scorecard in a different class, but it seems like you need to update the frames in scorecard class as well as your current game class, effectively duplicating the information. Can you not have a single array of frames record what the player has scored and all the classes could apply their logic on that one?